projects/kit/src/lib/masks/number/processors/clear-input-preprocessor.ts

```diff
@@ -0,0 +1,140 @@
+import type {MaskitoPreprocessor} from '@maskito/core';
+
+export function createClearInputPreprocessor({
+    decimalSeparator,
+    decimalZeroPadding,
+    prefix,
+    postfix,
+    minusSign,
+    precision,
+}: {
+    decimalSeparator: string;
+    decimalZeroPadding: boolean;
+    prefix: string;
+    postfix: string;
+    minusSign: string;
+    precision: number;
+}): MaskitoPreprocessor {
+    return ({elementState, data}, inputType) => {
+        if (
+            !decimalZeroPadding ||
+            !['deleteBackward', 'deleteForward'].includes(inputType)
+        ) {
+            return {elementState, data};
+        }
+
+        const {value, selection} = elementState;
+
+        const [start, end] = getSelectionWithoutPrefixPostfix({
+            value,
+            selection: [selection[0], selection[1]],
+            prefix,
+            postfix,
+        });
+
+        const resultStart = start + prefix.length;
+
+        const [integerPart, decimalPart = ''] = getIntegerDecimalParts({
+            selection: [start, end],
+            value: valueWithoutPrefixPostfix({
+                value: elementState.value,
+                prefix,
+                postfix,
+            }),
+            decimalSeparator,
+            precision,
+        });
+
+        return ((start === 0 && integerPart === '') ||
+            (start === minusSign.length && integerPart === minusSign)) &&
+            !Number(decimalPart)
+            ? {
+                  elementState: {
+                      value: prefix + integerPart + postfix,
+                      selection: [resultStart, resultStart],
+                  },
+                  data,
+              }
+            : {
+                  elementState: {
+                      value:
+                          prefix + integerPart + decimalSeparator + decimalPart + postfix,
+                      selection: [resultStart, resultStart],
+                  },
+                  data,
+              };
+    };
+}
+
+function getSelectionWithoutPrefixPostfix({
+    value,
+    selection,
+    prefix,
+    postfix,
+}: {
+    value: string;
+    selection: [number, number];
+    prefix: string;
+    postfix: string;
+}): [number, number] {
+    const cleanValue = valueWithoutPrefixPostfix({value, prefix, postfix});
+
+    let [start, end] = selection;
+
+    start = start <= prefix.length ? 0 : start - prefix.length;
+    end = end >= value.length - postfix.length ? cleanValue.length : end - prefix.length;
+
+    return [start, end];
+}
+
+function valueWithoutPrefixPostfix({
+    value,
+    prefix,
+    postfix,
+}: {
+    value: string;
+    prefix: string;
+    postfix: string;
+}): string {
```

Comment:
We already has the following utility:
https://github.com/taiga-family/maskito/blob/08bb9b32439ea1b05ecc3b24992061fa2732b152/projects/kit/src/lib/utils/extract-affixes.ts#L3-L10

Could it be used instead of creation new one ?